code/modules/modular_computers/hardware/job_disk.dm

```diff
@@ -229,3 +229,51 @@
 	name = "\improper FACEBUCKS disk"
 	icon_state = "cart-signal" // might need a new sprite
 	spam_delay = 5
+
+// maint spawn disks (including FACEBUCKS disk, and viruses)
+// Some of these are possibly inferior than its original version
+/obj/item/computer_hardware/hard_drive/role/random_maint_spawn
+	name = "random job disk"
+
+/obj/item/computer_hardware/hard_drive/role/random_maint_spawn/Initialize(mapload)
+	..()
+	var/static/disk_list = list(
+		/obj/item/computer_hardware/hard_drive/role/maint = 5, // value is pickweight
+		/obj/item/computer_hardware/hard_drive/role/medical = 1,
+		/obj/item/computer_hardware/hard_drive/role/chemistry = 1,
+		/obj/item/computer_hardware/hard_drive/role/signal/toxins = 1,
+		/obj/item/computer_hardware/hard_drive/role/security = 1,
+		/obj/item/computer_hardware/hard_drive/role/quartermaster = 1,
+		/obj/item/computer_hardware/hard_drive/role/virus/clown = 1,
+		/obj/item/computer_hardware/hard_drive/role/virus/mime = 1
+	)
+	var/static/adj_list = list("old", "weathered", "outdated")
+	var/obj/item/computer_hardware/hard_drive/role/chosen_disk = pickweight(disk_list)
+	chosen_disk = new chosen_disk(loc)
+	chosen_disk.disk_flags = NONE
+	switch(chosen_disk.type) // let's not make much subtype items
+		if(/obj/item/computer_hardware/hard_drive/role/medical)
+			chosen_disk.name = "\improper [pick(adj_list)] Med-U disk"
+			chosen_disk.disk_flags = DISK_MED // no robot access
+		if(/obj/item/computer_hardware/hard_drive/role/chemistry)
+			chosen_disk.name = "\improper [pick(adj_list)] ChemWhiz disk"
+			chosen_disk.disk_flags = DISK_CHEM // nothing different, but just in case
+		if(/obj/item/computer_hardware/hard_drive/role/signal/toxins)
+			chosen_disk.name = "\improper [pick(adj_list)] Signal Ace 2 disk"
+			chosen_disk.disk_flags = DISK_ATMOS | DISK_SIGNAL  // no chem
+		if(/obj/item/computer_hardware/hard_drive/role/security)
+			chosen_disk.name = "\improper [pick(adj_list)] R.O.B.U.S.T. disk"
+			chosen_disk.disk_flags = DISK_MANIFEST // only crew manifest
+		if(/obj/item/computer_hardware/hard_drive/role/quartermaster)
+			chosen_disk.name = "\improper [pick(adj_list)] space parts DELUXE disk"
+			chosen_disk.disk_flags = DISK_SILO_LOG // only silo log check
+		if(/obj/item/computer_hardware/hard_drive/role/virus/clown)
+			var/obj/item/computer_hardware/hard_drive/role/virus/temp = chosen_disk
+			temp.name = pickweight(list("\improper H.??. disk"=25, "\improper [pick(adj_list)] H.O.N.K. disk"=75))
+			temp.charges = 1
+		if(/obj/item/computer_hardware/hard_drive/role/virus/mime)
+			var/obj/item/computer_hardware/hard_drive/role/virus/temp = chosen_disk
+			temp.name = pickweight(list("\improper sound of ... disk"=25, "\improper [pick(adj_list)] sound of silence disk"=75))
+			temp.charges = 1
```

Comment:
These should either just be the hard drives as they are, or special subtypes instead of this